README.md

```diff
@@ -105,17 +105,14 @@ pip install hffs
 ...    first_row = root["embeddings/experiment_0"][0]
 ```
 
-* [`pyarrow`] + [`duckdb`]
+* [`duckdb`](https://duckdb.org/docs/guides/python/filesystems)
 
 ```python
 >>> import hffs
->>> import pyarrow.dataset as ds
 >>> import duckdb
 
 >>> fs = hffs.HfFileSystem("my-username/my-dataset-repo", repo_type="dataset")
-
->>> # Query a remote dataset and get the result as a dataframe
->>> dset = ds.dataset("data/", filesystem=fs)
->>> con = duckdb.connect()
->>> df = con.execute("SELECT * FROM dset LIMIT 1-").df()
+>>> duckdb.register_filesystem(fs)
```

Comment:
What if someone wants to register two HF repos ? Is it possible to pass the filesystem or the storage options at the same time as the query ?

Reply:
I have given thought to contributing a filesystem wrapper to fsspec that allows something like `hf+repo-name://`, then dispatching to various sub filesystems from there

Reply:
I think some users may be interested in playing with multiple HF filesystems (one per HF repository). I guess users would have to register + unregister their filesystems based on which repo they want to query.

> I have given thought to contributing a filesystem wrapper to fsspec that allows something like hf+repo-name://, then dispatching to various sub filesystems from there

I like the idea. Note that specifying `repo-name` alone may not be enough (you may instantiate two filesystems of the same repository, but at different git revisions !). I guess the extra fs identifier can be specifed by the user. Maybe something like this ?

```python
fs = HfFilesystem(..., fs_name="repo1")
path = "hf+repo1://data.parquet"
```

would be interested to know what you think @mariosasko

Reply:
IMO a better solution would be to require a (filesystem) class type as an argument rather than an instance in `duckdb.register_filesystem` (`fsspec.register_implementation` also accepts a class) and create an instance (of the previously registered filesystem) from a query instead.

And to address the issue with `storage_options`, perhaps a new function (e.g. `filesystem_url`) could be introduced (in DuckDB) that would allow passing custom storage options:
```python
duckdb.execute(
    "SELECT * FROM filesystem_url('hf://dataset/my-username/my-dataset:/data.parquet', STORAGE_OPTIONS={'revision': '...'}) LIMIT 10"
)  # e.g. similar to how `read_csv_auto` supports passing parameters
```

(Storage options are mostly simple Python types (int, string, bool), so it's possible to embed them within a query string, or replacements scans could be used as the most robust solution to allow passing local variables as storage options (maybe an overkill for now)).

Reply:
`register_filesystem` and `register_implementation` are doing very different things, so I'm not sure I agree with you there

```sql
SELECT * FROM filesystem_url('hf://dataset/my-username/my-dataset:/data.parquet', STORAGE_OPTIONS={'revision': '...'}) LIMIT 10
```
this syntax wouldn't work with how replacement scans currently work, you'd have to explicitly add the `read_parquet` call, or do an additional replacement scan inside the `filesystem_url` function, perhaps somehow detecting whether the result is being passed to a `read_*` function. It also means restating the storage options for every single sql query, which I'm not a fan of at all. Replacement scans for variables don't work inside libraries very well unfortunately

Reply:
Ok, makes sense.

What about making it possible to provide `storage_options` in `register_filesystem`. Then, `register_filesystem` would have the following signature:
```
register_filesystem(filesystem: Type[fsspec.AbstractFileSystem], storage_options: Optional[Dict] = None)
```

(the instance would then be created by merging an URL from the query and the filesystems `storage_options`)